src/plugins/passwd/CMakeLists.txt

```diff
@@ -1,20 +1,26 @@
+include (SafeCheckSymbolExists)
 include (LibAddMacros)
 
+set (plugin passwd)
+
 if (DEPENDENCY_PHASE)
-	try_compile (TEST_FGETPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasfgetpwent.c")
-	try_compile (TEST_PUTPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasputpwent.c")
+	add_definitions (-D_GNU_SOURCE -D_DARWIN_C_SOURCE)
+	safe_check_symbol_exists (fgetpwent "stdio.h;sys/types.h;pwd.h" TEST_FGETPWENT)
+	safe_check_symbol_exists (getline "stdio.h" TEST_GETLINE)
+	safe_check_symbol_exists (putpwent "stdio.h;sys/types.h;pwd.h" TEST_PUTPWENT)
 
-	if (TEST_PUTPWENT)
-		set (PUTPWENT 1)
-	else ()
-		set (PUTPWENT 0)
-	endif ()
 	if (TEST_FGETPWENT)
-		set (FGETPWENT 1)
+		add_definitions (-DUSE_FGETPWENT)
+	elseif (TEST_GETLINE)
+		add_definitions (-DUSE_FGETPWENT_LOCAL)
 	else ()
-		set (FGETPWENT 0)
+		remove_plugin (${plugin} "could not find required functions: fgetpwent || getline")
```

Comment:
I think it would make sense to use “or” instead of “||”. Probably everyone knows what the first term means, but only a few (programmers) know the second term. I would write something like:

> could not find one or both of the required functions: `fgetpwent`, `getline`

. Maybe it even makes sense to add one `remove_plugin` for each function to improve the output in case of a failure.

Reply:
> I think it would make sense to use “or” instead of “||”. 

agreed

> Maybe it even makes sense to add one remove_plugin for each function to improve the output in case of a failure.

That would imply that both are required right? But the implementation with getline is a replacement if fgetpwent can not be found hence the single message as you only need one of the functions.

Reply:
> hat would imply that both are required right?

Yep.

> But the implementation with getline is a replacement if fgetpwent can not be found hence the single message as you only need one of the functions.

In that case I would write something like:

> could not find `fgetpwent` nor `getline`

.